### ISSUE TITLE
noob-friendly options: set, settings and autoconnect

### ISSUE THREAD

**Issue by lsm5** [ivpn/desktop-app-cli#11]
# Feature request

## Description

The current CLI in my opinion takes some effort to get used to. I switched over from another VPN product and I notice that their CLI was much easier to get started with.

## Describe the solution you'd like
I guess Edward De must've forwarded these suggestions already, but since it's opensource, I thought why not file them here as well :)

So, I'm on:
```
$ rpm -q ivpn
ivpn-2.12.16-1.x86_64
```

and I think the following would be nice to have:

1. a 'settings' option which prints all the current settings.

2. a 'set' option to conveniently set/change any of the settings

3. an 'autoconnect' setting, and/or systemd unitfile with sane default installed by the rpm itself so
that I don't have to setup my own autoconnect script / crontab.

4. tab completion

5. Setting dns servers per interface and not Global / leave resolv.conf untouched.
Currently, my Global DNS server gets set to `w.x.y.z`. It would be nice if only wgivpn got set with that dns server while my `/etc/resolv.conf` was left unchanged.

```
$ resolvectl status
Global
         Protocols: LLMNR=resolve -mDNS -DNSOverTLS DNSSEC=no/unsupported
  resolv.conf mode: foreign                                              
Current DNS Server: w.x.y.z                                           
       DNS Servers: w.x.y.z                                           

Link 2 (eth0)
Current Scopes: none                                                        
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported

Link 3 (wlan0)
    Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6                                   
         Protocols: +DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
Current DNS Server: a.b.c.d                                               
       DNS Servers: a.b.c.d e.f.g.h i.j.k.l                    

Link 5 (wgivpn)
Current Scopes: none                                                        
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
```

## Misc
1. I noticed that the latest upstream release is not available as an rpm. Any ETA?

2. Where can I find the rpm spec file for Fedora? Also, do you have any plans to include it in Fedora by default? I could volunteer some of my time if that's something you're interested in.

**Comment by stenya** [comment]
@lsm5  thank you for the enhancement proposals.
I do not want to promise you that everything you suggested will be implemented. But definitely, this info will have an influence on our future decisions.

Few comments:
1. There are no so many settings for CLI. There is the only a default configuration for 'dns' and 'antitracker'. You can see by commands:
```
$ ivpn antitracker
Default config : AntiTracker Disabled
``` 
```
ivpn dns
Default config : Custom DNS 8.8.8.8
```

3. If you are able to use an IVPN graphical user interface -  there is the options `Launch at login` and `Autoconnect on launch`.

5. Setting DNS only for VPN interface potentially can lead to DNS leaks. That is why we using global configuration.

P.S. We just released the new client version (v3.3.7) for Linux with some fixes\improvements. All information on how to install it on Fedora, you can find here: https://www.ivpn.net/apps-linux/

**Comment by lsm5** [comment]
> @lsm5 thank you for the enhancement proposals.
> I do not want to promise you that everything you suggested will be implemented. But definitely, this info will have an influence on our future decisions.

Sure thing, SGTM.

> 
> Few comments:
> 
>     1. There are no so many settings for CLI. There is the only a default configuration for 'dns' and 'antitracker'. You can see by commands:
> 
> 
> ```
> $ ivpn antitracker
> Default config : AntiTracker Disabled
> ```
> 
> ```
> ivpn dns
> Default config : Custom DNS 8.8.8.8
> ```
> 
>     1. If you are able to use an IVPN graphical user interface -  there is the options `Launch at login` and `Autoconnect on launch`.

I guess I was speaking from the perspective of a cli-first user new to IVPN. I've gotten fairly comfortable with the ivpn cli to the point that I don't need noob settings anymore. So, if I was the only one to request such a thing, feel free to ignore this :)

> 
>     2. Setting DNS only for VPN interface potentially can lead to DNS leaks. That is why we using global configuration.

Ack, I see. SGTM.

> 
> 
> P.S. We just released the new client version (v3.3.7) for Linux with some fixes\improvements. All information on how to install it on Fedora, you can find here: https://www.ivpn.net/apps-linux/

Yup, using it now, works great so far. Btw, would you mind posting the rpm packaging source somewhere on github please? Can't seem to find it yet.

Feel free to close this if you wish. Thanks!

**Comment by stenya** [comment]
Hi @lsm5 ,

Here you can find the script which builds everything and creates rpm/deb packages ('base IVPN package'):
https://github.com/ivpn/desktop-app-cli/blob/master/References/Linux/build.sh

Instructions how to build IVPN 'base' package (deb/rpm): 
https://github.com/ivpn/desktop-app-cli#installation

Links to precompiled packages can be found here:
https://www.ivpn.net/apps-linux/#binaries
... and here:
https://github.com/ivpn/desktop-app-cli/blob/master/CHANGELOG.md
